web/models.py

```
@@ -2805,3 +2805,135 @@
+
+
+class MassClassViewer(models.Model):
+    """Model for tracking viewers in a mass class stream"""
+
+    user = models.ForeignKey(User, on_delete=models.CASCADE, related_name="mass_class_views")
+    stream = models.ForeignKey(MassClassStream, on_delete=models.CASCADE, related_name="viewers")
+    joined_at = models.DateTimeField(auto_now_add=True)
+    last_active = models.DateTimeField(auto_now=True)
+
+    class Meta:
+        unique_together = ["user", "stream"]
```

Ruff (0.8.2)

Mutable class attributes should be annotated with `typing.ClassVar`

(RUF012)